lib/matplotlib/tests/test_legend.py

```diff
@@ -250,6 +250,15 @@ def test_hatching():
     ax.legend(handlelength=4, handleheight=4)
 
 
+@image_comparison(baseline_images=['legend_arrow_annotation'])
+def test_legend_arrow_annotation():
+    # Related to issue 8236
+    # Tests arrow annotations showing up in legend
+    fig, ax = plt.subplots()
+    ax.annotate("foo", (0, 1), xytext=(2, 3), arrowprops={}, label='bar')
```

Comment:
please add:

ax.set_xtick_labels('')
ax.set_ytick_labels('')

This helps us cut down on extra text comparison failures.

Reply:
This test seems a bit minimal.  Might as well test all the new artists, as in your example in the PR (or a good subset of them).

Reply:
will do :)